.github/workflows/rust.yml

```diff
@@ -10,80 +15,78 @@ jobs:
       matrix:
         rust:
           - stable
-          - 1.53.0
+          - 1.63.0
           - nightly
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@master
         with:
-          profile: minimal
           toolchain: ${{ matrix.rust }}
-          override: true
       - name: Cargo update
         run: cargo update
-      - uses: actions-rs/cargo@v1
-        with:
-          command: check
+      - run: RUSTFLAGS="-D warnings" cargo check
 
   test:
     name: Test Suite
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: stable
-          override: true
-      - uses: actions-rs/cargo@v1
-        with:
-          command: test
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+      - run: cargo test
+      - run: cargo test --all-features
+      - run: cargo test --no-default-features
```

Comment:
There aren't any default features for this crate, just two optional features (`data` and `serialize`) - I thought it was best to include this coverage in the event that changes, but perhaps it's overkill. Happy to change either way.

Reply:
In this case, the features do not mix (they affect each a different part of the code), so it's good to keep it simple.
That said, I have an open question: would it be better to always reuse the same `rust.yml` layout in all crates? That would mostly mean using feature matrix for `check`, `test` and `clippy` commands.

Reply:
> would it be better to always reuse the same rust.yml layout in all crates? That would mostly mean using feature matrix for check, test and clippy commands.

I would be open to that! Is there a good "exemplar" to lift from? Perhaps `asn1-rs` and it's [test matrix](https://github.com/rusticata/asn1-rs/blob/26f6751c27faed9730614c2d3ca2ff7f42e00cd2/.github/workflows/rust.yml#L33-L42)?

Reply:
Added a commit to adopt a feature matrix.